relay-metrics/src/bucketing.rs

```diff
@@ -132,16 +133,24 @@ impl AggregatorConfig {
     fn get_flush_time(&self, bucket_timestamp: UnixTimestamp) -> Instant {
         let now = Instant::now();
 
-        let bucket_end = bucket_timestamp.to_instant() + self.bucket_interval();
-        let initial_flush = bucket_end + self.initial_delay();
-
-        if initial_flush > now {
-            // If the initial flush is still pending, use that.
-            initial_flush
-        } else {
-            // If the initial flush time has passed, debounce future flushes with the
-            // `debounce_delay` starting now.
-            now + self.debounce_delay()
+        match bucket_timestamp.to_instant() {
+            // For a bucket from the distance past, use debounce delay
+            UnboundedInstant::Earlier => now + self.debounce_delay(),
+            UnboundedInstant::Instant(instant) => {
+                let bucket_end = instant + self.bucket_interval();
+                let initial_flush = bucket_end + self.initial_delay();
+                if initial_flush > now {
+                    // If the initial flush is still pending, use that.
+                    initial_flush
+                } else {
+                    // If the initial flush time has passed, debounce future flushes with the
+                    // `debounce_delay` starting now.
+                    now + self.debounce_delay()
+                }
+            }
+            // Bucket from distant future. Best we can do is schedule it after initial delay
+            // TODO: or refuse metric altogether?
+            UnboundedInstant::Later => now + self.initial_delay(),
```

Comment:
@jan-auer Let me know if this makes sense. I'm unsure what to do about metrics with future timestamps.

Reply:
If the timestamp is not representable, then we're so far out of the valid ingestion range (something we've not yet implemented) that we can just drop the metric. WDYT if we just make this `Option<Instant>` everywhere and then bail out with an error to keep it simple?

Reply:
I think we need the ternary enum to support backdated metrics - at least if we want the tests to pass on macOS, where even the following panics:

```rust
use std::time::{Duration, Instant};

fn main() {
    let now = Instant::now();
    let then = now - Duration::from_secs(24 * 60 * 60);
    println!("{:?}", then);
}
```